### PR TITLE
Migration guide - list removed APIs with replacements

### PR DESCRIPTION
**Pull request by chriselion**
### Proposed change(s)
Fix link to migration guide, add list of changed APIs (from https://github.com/Unity-Technologies/ml-agents/pull/5024)

### Types of change(s)
- [x] Documentation update

### Checklist
- [x] Updated the [migration guide](https://github.com/Unity-Technologies/ml-agents/blob/main/docs/Migrating.md) (if applicable)

### CODE REVIEW

File: docs/Migrating.md [hunk 1]
```diff
@@ -1,25 +1,28 @@
 # Upgrading
 
-## :warning: Warning :warning:
```
Comment (chriselion) [inline docs/Migrating.md:3]: This hasn't been true since 1.0

File: Project/Packages/packages-lock.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
   "dependencies": {
     "com.unity.barracuda": {
-      "version": "1.3.3-preview",
+      "version": "1.4.0-preview",
```
Comment (vincentpierre) [inline Project/Packages/packages-lock.json:4]: This should be addressed in another PR.

**Review by vincentpierre — APPROVED** [review]

File: Project/Packages/packages-lock.json (hunk 2 above)
Comment (chriselion) [reply Project/Packages/packages-lock.json:4]: Yeah, accidentally added. Will remove (or merge latest).
